lib/appmap/hook/method.rb

```diff
@@ -45,42 +47,48 @@ def activate
         after_hook = self.method(:after_hook)
         with_disabled_hook = self.method(:with_disabled_hook)
 
-        hook_method_def = nil
-        hook_class.instance_eval do 
-          hook_method_def = Proc.new do |*args, &block|
-            instance_method = hook_method.bind(self).to_proc
-            call_instance_method = -> { instance_method.call(*args, &block) }
+        hook_method_def = Proc.new do |*args, &block|
+          instance_method = hook_method.bind(self).to_proc
+          call_instance_method = -> {
+            # https://github.com/applandinc/appmap-ruby/issues/153
+            if Util.ruby_minor_version >= 2.7 && args == ARRAY_OF_EMPTY_HASH && hook_method.arity == 1
```

Comment:
This is the actual fix for the bug. In this particular case, `instance_method.call(*args, &block)` will actually pass 0 args to the instance_method. We detect that case and make the correct call.